extension/entitystore/retryer.go

```diff
@@ -0,0 +1,103 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package entitystore
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"go.uber.org/zap"
+)
+
+const (
+	RequestLimitExceeded = "RequestLimitExceeded"
+	infRetry             = -1
+)
+
+var (
+	retryableErrorMap = map[string]bool{
+		"RequestLimitExceeded": true,
+	}
+)
+
+type Retryer struct {
+	oneTime         bool
+	retryAnyError   bool
+	successRetryMin int
+	successRetryMax int
+	backoffArray    []time.Duration
+	maxRetry        int
+	done            chan struct{}
+	logger          *zap.Logger
+}
+
+func NewRetryer(onetime bool, retryAnyError bool, successRetryMin int, successRetryMax int, backoffArray []time.Duration, maxRetry int, done chan struct{}, logger *zap.Logger) *Retryer {
+	return &Retryer{
+		oneTime:         onetime,
```

Comment:
nit: Why does a retryer have a `onetime` bool?

Reply:
Its a generic function to accommodate the scenario where only the value is retrieved once

Reply:
Non-blocking comment